crypto/crypto.go

```diff
@@ -158,37 +159,67 @@ func FromECDSAPub(pub *ecdsa.PublicKey) []byte {
 // HexToECDSA parses a secp256k1 private key.
 func HexToECDSA(hexkey string) (*ecdsa.PrivateKey, error) {
 	b, err := hex.DecodeString(hexkey)
-	if err != nil {
-		return nil, errors.New("invalid hex string")
+	if byteErr, ok := err.(hex.InvalidByteError); ok {
+		return nil, fmt.Errorf("invalid hex character %q in private key", byte(byteErr))
+	} else if err != nil {
+		return nil, errors.New("invalid hex data for private key")
 	}
 	return ToECDSA(b)
 }
 
 // LoadECDSA loads a secp256k1 private key from the given file.
 func LoadECDSA(file string) (*ecdsa.PrivateKey, error) {
-	stat, err := os.Stat(file)
+	fd, err := os.Open(file)
 	if err != nil {
 		return nil, err
 	}
-	size := stat.Size()
-	// Allow two extra chars for possible line ending to be checked later
-	if size < 64 || size > 66 {
-		return nil, fmt.Errorf("expected 64 bytes, got %v", size)
-	}
-	buf, err := ioutil.ReadFile(file)
+	defer fd.Close()
+
+	r := bufio.NewReader(fd)
+	buf := make([]byte, 64)
+	n, err := readASCII(buf, r)
```

Comment:
Instead of reading any ascii char, why not `readHexDigits()` that would stop when it encounters a non-hexdigit? Then we could give the exact location of the first wrong char and we don't have to complicate `HexToECDSA()`.

Reply:
The change in HexToECDSA is good to have on its own. It didn't report the offending character before, now it does.